crates/node/src/lib.rs

```diff
@@ -305,26 +305,28 @@ impl Node {
                 root_hash,
                 artifact,
             } => {
-                let identities = self.ctx_manager.get_context_owned_identities(context_id)?;
+                let possible_sending_key =
+                    self.ctx_manager.get_sender_key(&context_id, &author_id)?;
 
-                let Some(our_identity) = identities.into_iter().choose(&mut thread_rng()) else {
-                    bail!("no identities found for context: {}", context_id);
+                let sending_key = match possible_sending_key {
+                    Some(key) => key,
+                    None => todo!(), //initiate sync
```

Comment:
for example:

https://github.com/calimero-network/core/blob/db644002b9093b0363e2af55347ebb849161e07e/crates/node/src/lib.rs#L350-L354

I'd do

```rust
let Some(sender_key) = self
    .ctx_manager
    .get_sender_key(&context.id, &executor_public_key)?
else {
    return self.initiate_sync(context_id, source).await;
};
```

leave the rest of it to sync